package/run.ps1

```diff
@@ -46,6 +46,22 @@ function Get-StringHash {
     return (Get-FileHash -InputStream $stringAsStream -Algorithm SHA256).Hash.ToLower()
 }
 
+# stops the RKE2 process as well as other processes spawned from rke2
+function Stop-RKE2-Processes () {
+    $ProcessNames = @('rke2', 'kube-proxy', 'kubelet', 'containerd', 'calico-node')
+    foreach ($ProcessName in $ProcessNames) {
+        Write-LogInfo "Checking if $ProcessName process exists"
+        if ((Get-Process -Name $ProcessName -ErrorAction SilentlyContinue)) {
+            Write-LogInfo "$ProcessName process found, stopping now"
+            Stop-Process -Name $ProcessName
+            while (-Not(Get-Process -Name $ProcessName).HasExited) {
+                Write-LogInfo "Waiting for $ProcessName process to stop"
```

Comment:
Does this need to happen sequentially? Maybe we should trigger stopping all the processes and then checking whether they have stopped in a separate loop.

Reply:
also, should there be a limited number of retries before we return a failure or trying stopping again?

Reply:
Sounds good, I set a limit of 3 tries before failing - that gives each process 15 seconds to exit before we declare a failure

Reply:
I continued testing and realized that by not checking for process exits sequentially, the line `(Get-Process -Name $ProcessName).HasExited` can error out since the process will have been fully cleaned up. I'm updating the script to handle this scenario and to avoid the use of `.HasExited`